ports/servoshell/prefs.rs

```diff
@@ -181,393 +182,228 @@ pub fn read_prefs_map(txt: &str) -> HashMap<String, PrefValue> {
-                why
-            );
-        }
-    }
+    /// Wether or not to enable experimental web platform features.
+    #[bpaf(long, fallback(false))]
+    enable_experimental_web_platform_features: bool,
 
-    let layout_threads: Option<usize> = opt_match.opt_str("y").map(|layout_threads_str| {
-        layout_threads_str
-            .parse()
-            .unwrap_or_else(|err| args_fail(&format!("Error parsing option: -y ({})", err)))
-    });
-
-    let nonincremental_layout = opt_match.opt_present("i");
-
-    let random_pipeline_closure_probability = opt_match
-        .opt_str("random-pipeline-closure-probability")
-        .map(|prob| {
-            prob.parse().unwrap_or_else(|err| {
-                args_fail(&format!(
-                    "Error parsing option: --random-pipeline-closure-probability ({})",
-                    err
-                ))
-            })
-        });
+    // Exit after Servo has loaded the page and detected stable output image
+    #[bpaf(short('x'), long, fallback(false))]
+    exit: bool,
 
-    let random_pipeline_closure_seed =
-        opt_match
-            .opt_str("random-pipeline-closure-seed")
-            .map(|seed| {
-                seed.parse().unwrap_or_else(|err| {
-                    args_fail(&format!(
-                        "Error parsing option: --random-pipeline-closure-seed ({})",
-                        err
-                    ))
-                })
-            });
-
-    if opt_match.opt_present("devtools") {
-        let port = opt_match
-            .opt_str("devtools")
-            .map(|port| {
-                port.parse().unwrap_or_else(|err| {
-                    args_fail(&format!("Error parsing option: --devtools ({})", err))
-                })
-            })
-            .unwrap_or(preferences.devtools_server_port);
-        preferences.devtools_server_enabled = true;
-        preferences.devtools_server_port = port;
-    }
+    /// Exit on thread failure instead of displaying about:failure
+    #[bpaf(short('f'), long, fallback(false))]
+    hard_fail: bool,
 
-    let webdriver_port = opt_match.opt_default("webdriver", "7000").map(|port| {
-        port.parse().unwrap_or_else(|err| {
-            args_fail(&format!("Error parsing option: --webdriver ({})", err))
-        })
-    });
+    /// Headless mode
+    #[bpaf(short('z'), long, fallback(false))]
+    headless: bool,
 
-    let parse_resolution_string = |string: String| {
-        let components: Vec<u32> = string
-            .split('x')
-            .map(|component| {
-                component.parse().unwrap_or_else(|error| {
-                    args_fail(&format!("Error parsing resolution '{string}': {error}"));
-                })
-            })
-            .collect();
-        Size2D::new(components[0], components[1])
-    };
+    /// Wether or not to completely ignore certificate errors
+    #[bpaf(long, fallback(false))]
```

Comment:
:information_source: FWIW `long` is the default argument name representation. You can keep them for documentation purposes (I usually do) but just having something like this will give the same result.

```rust
/// Wether or not to completely ignore certificate errors
ignore_certificate_errors: bool,
```